actions/form/class.UserSettings.php

```
@@ -25,15 +25,39 @@
 use oat\oatbox\user\UserTimezoneServiceInterface;
 
 /**
- * This container initialize the settings form.
+ * This container initializes the settings form.
  *
  * @access public
  * @author Joel Bout, <joel.bout@tudor.lu>
  * @package tao
-
  */
 class tao_actions_form_UserSettings extends tao_helpers_form_FormContainer
 {
+    /** @var tao_models_classes_LanguageService */
+    private $languageService;
+
+    /** @var ServiceManager */
+    private $serviceManager;
+
+    /** @var UserTimezoneServiceInterface */
+    private $userTimezoneService;
+
+    public function __construct(array $data = [], array $options = [])
+    {
+        if (isset($options['LanguageService'])) {
+            $this->setLanguageService($options['LanguageService']);
+        }
+
+        if (isset($options['ServiceManager'])) {
+            $this->setServiceManager($options['ServiceManager']);
+        }
+
+        if (isset($options['UserTimezoneService'])) {
+            $this->setUserTimezoneService($options['UserTimezoneService']);
+        }
```

Maybe we can store this option names as constants?
E.g.
```php
public const OPTION_LANGUAGE_SERVICE = 'LanguageService';

// ...

if (isset($options[self::OPTION_LANGUAGE_SERVICE])) {
    $this->setLanguageService($options[self::OPTION_LANGUAGE_SERVICE]);
}
```

Also I'm thinking about provided values. If someone provides wrong value (string or something else) - setter will fail with wrong type exception.
```php
public const OPTION_LANGUAGE_SERVICE = 'LanguageService';

// ...

if (($options[self::OPTION_LANGUAGE_SERVICE] ?? null) instanceof tao_models_classes_LanguageService) {
    $this->setLanguageService($options[self::OPTION_LANGUAGE_SERVICE]);
}
```

---

Adding the option names as constants sounds like a good idea for me, changed here: e755569b7e0f7fb0c0c38e4baa8e7c95baf2b5b4 Thanks!

For the second part... I'm not sure if we want to mask an error caused by providing wrong data types.

I would prefer to _let it crash_ in this case, as that would be the case (throwing a `TypeError`, as you already pointed out) if we had proper constructor parameters for them and they were directly passed instead of having them as array elements. Silent errors are tricky to diagnose later :worried: 

However, an intermediate approach would be to let wrong types here but log an error in that case, with the tradeoff of coupling this to the logger.

What do you think is best here? Crash, ignore, or ignore+logError?

Note this is a legacy class, but it seems right now it is only called from our `UserSettingsFormFactory` class.

cc/ @gabrielfs7

---

`instanceof` checks added in 2ee916525e1472405ff3076cbbda66d53f2afe59